File: app/src/main/java/tk/wasdennnoch/scoop/view/CrashRecyclerView.java
package tk.wasdennnoch.scoop.view;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class CrashRecyclerView extends RecyclerView {

    private LinearLayoutManager mManager;

    public CrashRecyclerView(Context context) {
        this(context, null);
    }

    public CrashRecyclerView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public CrashRecyclerView(Context context, @Nullable AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        mManager = new LinearLayoutManager(context);
        setLayoutManager(mManager);
    }

    public void setReverseOrder(boolean reverse) {
        mManager.setReverseLayout(reverse);
        mManager.setStackFromEnd(reverse);
    }

}

File: app/src/main/java/taco/scoop/ui/view/CrashRecyclerView.kt
package taco.scoop.ui.view

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class CrashRecyclerView @JvmOverloads constructor(
    context: Context?,
    attrs: AttributeSet? = null,
    defStyle: Int = 0
) : RecyclerView(
    context!!, attrs, defStyle
) {
    private val mManager: LinearLayoutManager = LinearLayoutManager(context)
    fun setReverseOrder(reverse: Boolean) {
        mManager.reverseLayout = reverse
        mManager.stackFromEnd = reverse
    }

    init {
        layoutManager = mManager
    }
}
